src/main/java/org/springframework/integration/aws/lock/DynamoDbLockRegistry.java

```diff
@@ -467,11 +467,11 @@ public boolean tryLock(long time, TimeUnit unit) throws InterruptedException {
 				return false;
 			}
 
-			long additionalTimeToWait = TimeUnit.MILLISECONDS.convert(time, unit) - System.currentTimeMillis() + start;
+			long additionalTimeToWait = Math.max(TimeUnit.MILLISECONDS.convert(time, unit) - System.currentTimeMillis() + start, 0L);
 
 			this.acquireLockOptionsBuilder
 					.withAdditionalTimeToWaitForLock(additionalTimeToWait)
-					.withRefreshPeriod(0L);
+					.withRefreshPeriod(DEFAULT_REFRESH_PERIOD_MS);
```

Comment:
Why `default`? I believe we can simply rely on the provided `refreshPeriod`, which frankly can be configured to `0` 😄

Reply:
That was my first approach but then I was thinking that it a bit tricky to find a value for `refreshValue` that both satisfy `lock()` and `tryLock()`. 

In the first case, we might want to want to check at longer intervals that in the second one. For example, in `lock()` you might want a `refreshPeriod` longer than the `leaseDuration` while in `tryLock()`, you want it shorter.

That being said, I won't insist and can revert back my second commit if you prefer, your choice!

Reply:
Now I think this is already a target client detail and it doesn't what contract is on the top `lock()` or `tryLock()`.
So, if there is a requirement to have such a `refreshPeriod`, we don't have choice unless follow it and expose such an option for target users to adjust according their requirements.
I even really believe that we can't have it less than 1 second. Otherwise we are going to end up with something like `ProvisionedThroughputExceededException`.

So, my point is let's rely on the `this.refreshPeriod` property value everywhere we need `withRefreshPeriod()`!

Thank you for sharing your thoughts!